tests/ci/finish_check.py

```diff
@@ -20,84 +20,103 @@
-        if has_failure:
-            set_mergeable_check(commit, "workflow failed", "failure")
+        # in MQ Mergeable check status must never be green if any failures in the workflow
+        if has_workflow_failures:
+            set_mergeable_check(commit, "workflow failed", FAILURE)
         else:
             # This must be the only place where green MCheck is set in the MQ (in the end of CI) to avoid early merge
-            set_mergeable_check(commit, "workflow passed", "success")
-    else:
-        statuses = get_commit_filtered_statuses(commit)
-        state = trigger_mergeable_check(commit, statuses, set_if_green=True)
-
-        # Process upstream StatusNames.SYNC
-        if (
-            pr_info.head_ref.startswith(f"{SYNC_BRANCH_PREFIX}/pr/")
-            and GITHUB_REPOSITORY != GITHUB_UPSTREAM_REPOSITORY
-        ):
-            upstream_pr_number = int(pr_info.head_ref.split("/pr/", maxsplit=1)[1])
-            update_upstream_sync_status(
-                upstream_pr_number,
-                pr_info.number,
-                gh,
-                state,
-                can_set_green_mergeable_status=True,
-            )
-
-        ci_running_statuses = [s for s in statuses if s.context == StatusNames.CI]
-        if not ci_running_statuses:
-            return
-        # Take the latest status
-        ci_status = ci_running_statuses[-1]
-
-        has_failure = False
-        has_pending = False
-        for status in statuses:
-            if status.context in (StatusNames.MERGEABLE, StatusNames.CI):
-                # do not account these statuses
-                continue
-            if status.state == PENDING:
-                if status.context == StatusNames.SYNC:
-                    # do not account sync status if pending - it's a different WF
-                    continue
-                has_pending = True
-            elif status.state == SUCCESS:
+            set_mergeable_check(commit, "workflow passed", SUCCESS)
+        return
+
+    statuses = get_commit_filtered_statuses(commit)
+    state = trigger_mergeable_check(commit, statuses, set_if_green=True)
+
+    # Process upstream StatusNames.SYNC
+    if (
+        pr_info.head_ref.startswith(f"{SYNC_BRANCH_PREFIX}/pr/")
+        and GITHUB_REPOSITORY != GITHUB_UPSTREAM_REPOSITORY
+    ):
+        upstream_pr_number = int(pr_info.head_ref.split("/pr/", maxsplit=1)[1])
+        update_upstream_sync_status(
+            upstream_pr_number,
+            pr_info.number,
+            gh,
+            state,
+            can_set_green_mergeable_status=True,
+        )
+
+    ci_running_statuses = [s for s in statuses if s.context == StatusNames.CI]
+    if not ci_running_statuses:
+        return
+    # Take the latest status
+    ci_status = ci_running_statuses[-1]
+
+    has_failure = False
+    has_pending = False
+    error_cnt = 0
+    for status in statuses:
+        if status.context in (StatusNames.MERGEABLE, StatusNames.CI):
```

Comment:
```suggestion
        if status.context in (StatusNames.MERGEABLE, StatusNames.CI, StatusNames.SYNC):
```

The SYNC should be out of the scope in any case, right?

Reply:
i don't mind "sync" in the report

Reply:
added